### PR TITLE
Remove experimental async functions

### PR DESCRIPTION
**Pull request by mhegazy**
Now that async/await support has reached stage 3 in the TC39 proposal process, time to remove the `--experimentalAsyncFunctions` flag

### CODE REVIEW

**Comment by yuit** [comment]
lgtm.

File: src/compiler/program.ts [hunk 1]
```diff
@@ -1076,11 +1076,6 @@ namespace ts {
                 !options.experimentalDecorators) {
                 programDiagnostics.add(createCompilerDiagnostic(Diagnostics.Option_0_cannot_be_specified_without_specifying_option_1, "emitDecoratorMetadata", "experimentalDecorators"));
             }
-
-            if (options.experimentalAsyncFunctions &&
-                options.target !== ScriptTarget.ES6) {
```
Comment (sandersn) [inline src/compiler/program.ts]: do we now have ES5 support for async functions? What happens if you target ES5 and then use an async function?

Comment (mhegazy) [inline src/compiler/program.ts]: there should be another message `Async functions are only available when targeting ECMAScript 6 and higher.` which i have not changed

**Comment by sandersn** [comment]
:+1:

**Comment by ctaggart** [comment]
I had to lookup and see what TC39 was. It is a well documented here:
http://tc39.github.io/ecmascript-asyncawait/

**Comment by elhigu** [comment]
https://github.com/Microsoft/TypeScript/wiki/Compiler-Options needs an update... this flag is still mentioned there without deprecation note.

**Comment by NekR** [comment]
Is there a way to use async/await but still target ES5? I mean, there is many envs which do not support ES2015 100%, but they support `yield`. So would be good to target ES5, but still use `yield` and async/await.

**Comment by DanielRosenwasser** [comment]
For the time being, no. But we're working on supporting async/await in pre-ES6 targets.

**Comment by NekR** [comment]
@DanielRosenwasser thanks. Just got the idea:
`TS -> ES6` and then transpiling to ES5 via Babel with disabled `yield` transpilation.

**Pull request merged**